<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
    <title>Dr. Simon Rogers</title>

    <!-- Bootstrap -->
    <link href="css/bootstrap.min.css" rel="stylesheet">
    <link href="custom.css" rel="stylesheet">
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
  </head>
<body id="page-top" data-spy="scroll" data-target=".navbar-fixed-top">

  <nav class="navbar navbar-default navbar-fixed-top">
    <div class="container-fluid">
      <div class="navbar-header">
        <a class="navbar-brand" href="#">
          <!-- <img alt="Brand" src="images/UoG.png" width=50px> -->
        </a>
      </div>

      <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
            <li class="active"><a href="#home">Home</a></li>
            <li><a href="#about">Biography</a></li>
            <li><a href="#research">Research</a></li>
            <li><a href="#students">PhD Students</a></li>
            <li><a href="#teaching">Teaching</a></li>
            <li><a href="#fcml">FCML</a></li>
            <li><a href="#direct">Directions</a></li>
          </ul>
      </div>
    </div>
  </nav>

  
  <div class="content-section" id="home" style="padding-top:150px;">
    <div class="container">
      <div class="row">
        <div class="col-md-3">
          <img vspace=50 width=200px src="images/UoG_colour.png">
        </div>
        <div class="col-md-6" style="text-align:right;">
          <h2>Dr. Simon Rogers</h2>
          <p>School of Computing Science, University of Glasgow</p>
          <p>S123, Sir Alwyn Williams Building, G12 8RZ, UK</p>
          <p>0141 330 1649</p>
          <p><a href="mailto:simon.rogers@glasgow.ac.uk">simon.rogers@glasgow.ac.uk</a></p>
          <p><a href="http://www.gla.ac.uk/schools/computing/staff/simonrogers/">Official University of Glasgow page</a>
        </div>
        <div class="col-md-3">
          <img vspace=30 width=200px src="images/sdr.jpg">
        </div>
      </div>
    </div>
    <hr />
  </div>


  <div class="content-section" id="about">
    <div class="container">
      <div class="row">
        <div class="col-md-8">
        <h2>Biography</h2>
          <div style="text-align:left;">
            <p>I'm a senior lecturer in the School of Computing Science at the University of Glasgow. Before starting a permanent post in Glasgow I did a couple of Post Doctoral positions (under <a href="http://www2.warwick.ac.uk/fac/sci/statistics/staff/academic-research/girolami/">Mark Girolami</a>) and prior to that I did my Undergraduate degree (Electrical and Electronic Engineering) and PhD (<i><a href="files/srogers_thesis.pdf">Machine Learning Techniques for Microarray Analysis</a></i>, supervisor <a href="http://seis.bris.ac.uk/~enicgc/">Colin Campbell</a>) at the University of Bristol.</p>

            <p>My research involves the development of Machine Learning and Statistical techniques to help with the analysis of complex datasets, particularly within the field of Metabolomics but also other -omics fields, Human-Computer Interaction and Information Retreival. My Metabolomics work is done in collaboration with many people, particularly <a href="http://www.polyomics.gla.ac.uk">Glasgow Polyomics</a> (of which I'm an affiliate member).</p>

            <p>In 2011, Mark Girolami and I published a textbook: <i><a href="#fcml">A First Course in Machine Learning</a></i> which you might find useful if you want an introduction to modern (fairly probabilistic) Machine Learning but don't know a great deal of maths. A second edition with additional material (Gaussian Processes, Dirichlet Processes, was published in August 2016.</p>

            <p>I am always interested in hearing from <b>potential PhD candidates</b>. If any of the research below interests you, please get in touch. If you want to find my office, there are some instructions <a href="#direct">below</a>.</p>

            <p>I vary rarely blog, but when I do it is at <a href="https://sirogers.wordpress.com">sirogers.wordpress.com</a>. You can find me on <b>github</b> as <a href="https://github.com/sdrogers">sdrogers</a>, my <b>orcid</b> is <a href="http://orcid.org/0000-0003-3578-4477">orcid.org/0000-0003-3578-4477</a> and if you like <b>Google Scholar</b>, I'm <a href="https://scholar.google.co.uk/citations?user=5EiC_wEAAAAJ&hl=en">here</a>.</p>

            <p>I review for many journals and conferences, as well as being an associate editor for PLoS One. I have acted as an internal and external examiner for several PhD theses (in the UK and abroad) and am currently one of the external examiners for the computing science taught masters programmes at the University of Manchester.

            <p>I currently hold various non-research roles within the School of Computing Science: Graduate School Convenor (aka PG tutor), deputy senior adviser of studies and I'm a member of the College of Science and Engineering's appeals committee. I played a substantial role in the successful Athena Swan bronze application made by the School.</p> 

          </div>
        </div>
        <div class="col-md-4">
          <a class="twitter-timeline" href="https://twitter.com/sdrogers" data-widget-id="367216387506176001">Tweets by @sdrogers</a> <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
        </div>
      </div>
    </div>
  </div>

  <hr />
  

  <div class="content-section" id="research">
    <div class="container">
      <div class="row">
        <div class="col-md-12">
          <h2>Research</h2>
        </div>
      </div>

      <div class="row">
        <div class="col-md-12">
          <p>A full list of publications is available at Glasgow Universities online repository, <a href="http://eprints.gla.ac.uk/view/author/11628.html">Enlighten</a>. You can also find a large selection of my publications on <a href="http://orcid.org/0000-0003-3578-4477">orcid</a>, as well as a list of funding (my ID is 0000-0003-3578-4477), or <a href="https://scholar.google.co.uk/citations?user=5EiC_wEAAAAJ&hl=en">Google Scholar</a>. If you would like copies of anything, <a href="mailto:simon.rogers@glasgow.ac.uk">email me</a>.</p>
        </div>
      </div>

      <div class="row">
        <div class="col-md-4">
          <img src="images/metassignsmall.png" width=100%>
          <img src="images/network_cartoon.png" width=100%>
        </div>
        <div class="col-md-8">
          <h3>Computational Metabolomics</h3>
          <p>My main research activities are in the application of machine learning and applied statistical techniques to problems in metabolomics. Metabolomics is the study of the small molecule content of an organism and I'm particularly interested in the analysis of mass spectrometry metabolomics data. Mass Spectrometry allows, in theory, the simultanoues measurement of the abundance of many metabolites but use of this technology is hindered by the difficuly in analysising the data, particularly in identifying the molecules present in the data from the measured masses.</p>
          <p>Some notable contributions in this area include <a href="http://dx.doi.org/10.1093/bioinformatics/btn642">a method for using the dependencies between metabolites to aid in identification</a>, a <a href="http://dx.doi.org/10.1093/bioinformatics/btu370">probabilistic method for metabolite annotation / identification</a>, and the use of peak groupe to <a href="http://dx.doi.org/10.1093/bioinformatics/btv072">improve alignment</a> and <a href="http://dx.doi.org/10.1093/bioinformatics/btu455">differential expression computation.</a><p>
          <p>Currently, I'm working on novel ways of analysing molecular fragmentation data.</p>
          <p>Much of this work is done in collaboration with <a href="http://www.polyomics.gla.ac.uk">Glasgow Polyomics</a>, of which I'm an affiliate member.</p>
        </div>
      </div>

      <div class="row">
        <div class="col-md-8">
          <h3>Other Computational Biology</h3>
          <p><b>Microarray analysis:</b>&nbsp;The research in my PhD centered around developing Machine Learning techniques to analyse data from <a href="https://en.wikipedia.org/wiki/Microarray" target="new">cDNA microarrays</a>, including a <a href="http://dx.doi.org/10.1089/106652703321825928" target="new">2003 paper</a> on the number of arrays needed to be able to classify diseased from healthy samples which is still being cited. I also performed one of the first analyses on <a href="http://dx.doi.org/10.1093/bioinformatics/btn553">combine time series transcriptomic and proteomic data</a>.</p>
          <p><b>Parameter inference:</b>&nbsp;I am currently involved in an EPSRC funded project (EP/L020319/1) on parameter inference in mechanistic models of biological systems (with Dirk Husmeier and Maurizio Filippone) where we are interested in using gradient matching to speed up inference in systems of ordinary differential equations (<a href="http://eprints.gla.ac.uk/119575/">ICML 2016</a>) as well as a BBSRC funded project (BB/L001276/1) on stomatal dynamics with Mike Blatt (plant sciences, Glasgow) and others.</p>
          <p><b>Leaukaemia:</b>&nbsp;I co-supervise two PhD students who are using bioinformatics tools to help learn about Chronic Myeloid Leukaemia, with <a href="http://www.gla.ac.uk/researchinstitutes/cancersciences/staff/tessaholyoake/">Prof. Tessa Holyoake</a>.</p>
        </div>
        <div class="col-md-4">
          <img src="images/cb.png" width=100%>
          <img src="images/array.png" width=100%>
        </div>
      </div>


      <div class="row">
        <div class="col-md-4">
          <img src="images/gptype.png" width=100%>
          <img src="images/hands.png" width=100%>
        </div>
        <div class="col-md-8">
          <h3>Human-Computer Interaction</h3>
          <p>I'm interested in how we can better model the <i>uncertainty</i> in user interactions with a goal towards making better systems. This work began with the work on <a href="http://eprints.gla.ac.uk/34252/">FingerCloud</a>, which used a particle filter to model the position of the finger in 3D above an array of capacitive sensors. This filter was used to control a map, where the less engaged the user was the more autonomous the map became. We extended this to <a href="http://eprints.gla.ac.uk/56235/">AnglePose</a> which inferred the pitch and yaw angles of the interacting finger, along with its position. This work was taken forward in <a href="http://theses.gla.ac.uk/6318/">Daryl Weir's</a> PhD in which he looked at modelling uncertainty in touch. This started with a UIST paper that used Machine Learning to <a href="http://eprints.gla.ac.uk/68064/">learn offset models for specific users</a> and ended with a <a href="http://eprints.gla.ac.uk/89637/">CHI paper</a> demonstrating a keyboard that used both the learned offset and variability for specific users to improve typing performance. The offset work was built upon by <a href="http://www.medien.ifi.lmu.de/team/daniel.buschek/">Daniel Buschek</a>, who published an interesting study in <a href="https://www.google.co.uk/url?sa=t&rct=j&q=&esrc=s&source=web&cd=5&cad=rja&uact=8&ved=0ahUKEwiejZntzOfOAhVCExoKHQAoBAgQFgg8MAQ&url=http%3A%2F%2Feprints.gla.ac.uk%2F80621%2F&usg=AFQjCNHb6CwsRerA_GTIe6gqxMqHnGs_Hg&sig2=UPLYmZNkUgSReQLKZZA1eg">MobileHCI</a> investigating how offset patterns varied across users and varied for the same user on different devices.</p>
          <p>With John Williamson, I have been co-supervising Faizuddin M.Noor's PhD on using back of device sensors for implicit interactions. We have shown that it is possible to detect where a user is going to touch a screen before they do so (<A href=" http://dx.doi.org/10.1145/2556288.2557148">CHI 2014</A>) and, more recently showed that it is possible to detect user errors due to hand movement and changes in their grip patterns (<A href="http://eprints.gla.ac.uk/114779/">CHI 2016</A>).
        </div>
      </div>

      <div class="row">
        <div class="col-md-12">
          <h3>Other</h3>
          <div style="text-align:left;">
            <p><b>Information Retreival</b>: Along with Iadh Ounis, I supervised Edwin Thuma's PhD on <i>A semi-automated FAQ retrieval system for HIV/AIDS</i>. Edwin built a system to which users could send an SMS including questions about HIV/AIDS that would attempt to send an answer from a FAQ handbook used by the department of health in Botswana. There were three publications arising from this thesis: <a href="http://eprints.gla.ac.uk/78673/">Exploiting query logs and field-based models...</a>, <a href="http://eprints.gla.ac.uk/89123/">Detecting missing content queries...</a>, and <a href="http://eprints.gla.ac.uk/76925/">Evaluating bad query abandonment...</a>
            </p>
            <p><b>Robotics</b>: With Paul Siebert, I currently supervise Li Sun's PhD on using a robot to manipulate clothing.</p>
          </div>
          <!-- <div class="col-md-4">
        </div> -->
        </div>
      </div>


      <div class="row">
        <div class="col-md-8">
          <h3>Older things</h3>
          <div style="text-align:left;">
          <p>Code and data for older work can be found at the following links:</p>
          <ul>
            <li><a href="http://www.dcs.gla.ac.uk/inference/metsamp">Metabolomic Sampler</a></li>
            <li><a href="http://www.dcs.gla.ac.uk/inference/genomic_integration">Combining mRNA and protein profiles</a></li>
            <li><a href="old/lpd/lpd.html">Latent Process Decomposition</a></li>
            <li><a href="old/kernel_comb/kernel_comb.html">Kernel Combinations</a></li>
            <li><a href="old/nets/reg_nets.html">Regulatory Networks</a></li>
            <li><a href="old/tfa/index.html">Transcription Factor Activity</a></li>
          </ul>
          </div>
        </div>
        <div class="col-md-4">
        </div>
      </div>



    </div>
  </div>
  <hr />
  <div class="content-section" id="students">
    <div class="container">
      <div class="row">
        <div class="col-md-12">
        <h2>PhD Students</h2>
        </div>
      </div>
      <div class="row" style="text-align:left;">
        <div class="col-md-4">
          <h3>Current Students</h3>
            <div>
              <p><b>Joe Wandy</b> <i>Bayesian methods for metabolomics</i></p>
              <p><b>Faizuddin M.Noor</b> <i>Implicit interactions for mobile devices</i></p>
              <p><b>Lorna Jackson</b></p>
              <p><b>Eduardo Gómez Castañeda</b></p>
            </div>
        </div>
        <div class="col-md-8">
          <h3>Alumni</h3>
            <div>
              <p><b>Dr. Daryl Weir</b> <i><a href="http://theses.gla.ac.uk/6318/">Modelling Uncertainity in Touch Interactions</a></i>
              <button class="btn btn-xs" data-toggle="collapse" data-target="#daryl">toggle abstract</button></p>
              <p id="daryl" class="collapse" style="text-align:left;">
              Touch interaction is an increasingly ubiquitous input modality on modern devices. It appears on devices including phones, tablets, smartwatches and even some recent laptops. Despite its popularity, touch as an input technology suffers from a high level of measurement uncertainty. This stems from issues such as the ‘fat finger problem’, where the soft pad of the finger creates an ambiguous contact region with the screen that must be approximated by a single touch point. In addition to these physical uncertainties, there are issues of uncertainty of intent when the user is unsure of the goal of a touch. Perhaps the most common example is when typing a word, the user may be unsure of the spelling leading to touches on the wrong keys. The uncertainty of touch leads to an offset between the user’s intended target and the touch position recorded by the device. While numerous models have been proposed to model and correct for these offsets, existing techniques in general have assumed that the offset is a deterministic function of the input. We observe that this is not the case — touch also exhibits a random component. We propose in this dissertation that this property makes touch an excellent target for analysis using probabilistic techniques from machine learning. These techniques allow us to quantify the uncertainty expressed by a given touch, and the core assertion of our work is that this allows useful improvements to touch interaction to be obtained. We show this through a number of studies. In Chapter 4, we apply Gaussian Process regression to the touch offset problem, producing models which allow very accurate selection of small targets. In the process, we observe that offsets are both highly non-linear and highly user-specific. In Chapter 5, we make use of the predictive uncertainty of the GP model when applied to a soft keyboard — this allows us to obtain key press probabilities which we combine with a language model to perform autocorrection. In Chapter 6, we introduce an extension to this framework in which users are given direct control over the level of uncertainty they express. We show that not only can users control such a system succesfully, they can use it to improve their performance when typing words not known to the language model. Finally, in Chapter 7 we show that users’ touch behaviour is significantly different across different tasks, particularly for typing compared to pointing tasks. We use this to motivate an investigation of the use of a sparse regression algorithm, the Relevance Vector Machine, to train offset models using small amounts of data.
              </p>


              <p><b>Dr. Rebecca Mancy</b> <i><a href="http://theses.gla.ac.uk/6219/">Modelling persistence in spatially-explicit ecological and epidemiological systems</a></i>
              <button class="btn btn-xs" data-toggle="collapse" data-target="#rebecca">toggle abstract</button></p>
              <p id="rebecca" class="collapse" style="text-align:left;">
              In this thesis, we consider the problem of long-term persistence in ecological and epidemiological systems. This is important in conservation biology for protecting species at risk of extinction and in epidemiology for reducing disease prevalence and working towards elimination. Understanding how to predict and control persistence is critical for these aims. In Chapter 2, we discuss existing ways of characterising persistence and their relationship with the modelling paradigms employed in ecology and epidemiology. We note that data are often limited to information on the state of particular patches or populations and are modelled using a metapopulation approach. In Chapter 3, we define persistence in relation to a pre-specified time horizon in stochastic single-species and two-species competition models, comparing results between discrete and continuous time simulations. We find that discrete and continuous time simulations can result in different persistence predictions, especially in the case of inter-specific competition. The study also serves to illustrate the shortcomings of defining persistence in relation to a specific time horizon. A more mathematically rigorous interpretation of persistence in stochastic models can be found by considering the quasi-stationary distribution (QSD) and the associated measure of mean time to extinction from quasi-stationarity. In Chapter 4, we investigate the contribution of individual patches to extinction times and metapopulation size, and provide predictors of patch value that can be calculated easily from readily available data. In Chapter 5, we focus directly on the QSD of heterogeneous systems. Through simulation, we investigate possible compressions of the QSD that could be used when standard numerical approaches fail due to high system dimensionality, and provide guidance on appropriate compression choices for different purposes. In Chapter 6, we consider deterministic models and investigate the effect of introducing additional patch states on the persistence threshold. We suggest a possible model that might be appropriate for making predictions that extend to stochastic systems. By considering a family of models as limiting cases of a more general model, we demonstrate a novel approach for deriving quantities of interest for linked models that should help guide modelling decisions. Finally, in Chapter 7, we draw out implications for conservation biology and disease control, as well as for future work on biological persistence.
              </p>

              <p><b>Dr. Edwin Thuma</b> <i><a href="http://theses.gla.ac.uk/6280/">A semi-automated FAQ retrieval system for HIV/AIDS</a></i>
              <button class="btn btn-xs" data-toggle="collapse" data-target="#edwin">toggle abstract</button></p>
              <p id="edwin" class="collapse" style="text-align:left;">
              This thesis describes a semi-automated FAQ retrieval system that can be queried by users through short text messages on low-end mobile phones to provide answers on HIV/AIDS related queries. First we address the issue of result presentation on low-end mobile phones by proposing an iterative interaction retrieval strategy where the user engages with the FAQ retrieval system in the question answering process. At each iteration, the system returns only one question-answer pair to the user and the iterative process terminates after the user's information need has been satisfied. Since the proposed system is iterative, this thesis attempts to reduce the number of iterations (search length) between the users and the system so that users do not abandon the search process before their information need has been satisfied. Moreover, we conducted a user study to determine the number of iterations that users are willing to tolerate before abandoning the iterative search process. We subsequently used the bad abandonment statistics from this study to develop an evaluation measure for estimating the probability that any random user will be satisfied when using our FAQ retrieval system. In addition, we used a query log and its click-through data to address three main FAQ document collection deficiency problems in order to improve the retrieval performance and the probability that any random user will be satisfied when using our FAQ retrieval system. Conclusions are derived concerning whether we can reduce the rate at which users abandon their search before their information need has been satisfied by using information from previous searches to: Address the term mismatch problem between the users' SMS queries and the relevant FAQ documents in the collection; to selectively rank the FAQ document according to how often they have been previously identified as relevant by users for a particular query term; and to identify those queries that do not have a relevant FAQ document in the collection. In particular, we proposed a novel template-based approach that uses queries from a query log for which the true relevant FAQ documents are known to enrich the FAQ documents with additional terms in order to alleviate the term mismatch problem. These terms are added as a separate field in a field-based model using two different proposed enrichment strategies, namely the Term Frequency and the Term Occurrence strategies. This thesis thoroughly investigates the effectiveness of the aforementioned FAQ document enrichment strategies using three different field-based models. Our findings suggest that we can improve the overall recall and the probability that any random user will be satisfied by enriching the FAQ documents with additional terms from queries in our query log. Moreover, our investigation suggests that it is important to use an FAQ document enrichment strategy that takes into consideration the number of times a term occurs in the query when enriching the FAQ documents. We subsequently show that our proposed enrichment approach for alleviating the term mismatch problem generalise well on other datasets. Through the evaluation of our proposed approach for selectively ranking the FAQ documents, we show that we can improve the retrieval performance and the probability that any random user will be satisfied when using our FAQ retrieval system by incorporating the click popularity score of a query term t on an FAQ document d into the scoring and ranking process. Our results generalised well on a new dataset. However, when we deploy the click popularity score of a query term t on an FAQ document d on an enriched FAQ document collection, we saw a decrease in the retrieval performance and the probability that any random user will be satisfied when using our FAQ retrieval system. Furthermore, we used our query log to build a binary classifier for detecting those queries that do not have a relevant FAQ document in the collection (Missing Content Queries (MCQs))). Before building such a classifier, we empirically evaluated several feature sets in order to determine the best combination of features for building a model that yields the best classification accuracy in identifying the MCQs and the non-MCQs. Using a different dataset, we show that we can improve the overall retrieval performance and the probability that any random user will be satisfied when using our FAQ retrieval system by deploying a MCQs detection subsystem in our FAQ retrieval system to filter out the MCQs. Finally, this thesis demonstrates that correcting spelling errors can help improve the retrieval performance and the probability that any random user will be satisfied when using our FAQ retrieval system. We tested our FAQ retrieval system with two different testing sets, one containing the original SMS queries and the other containing the SMS queries which were manually corrected for spelling errors. Our results show a significant improvement in the retrieval performance and the probability that any random user will be satisfied when using our FAQ retrieval system.
              </p>



              <p><b>Dr. Mikhail Churakov</b> <i><a href="http://theses.gla.ac.uk/6417/">Spatial and network aspects of the spread of infectious diseases in livestock populations</a></i>
              <button class="btn btn-xs" data-toggle="collapse" data-target="#mikhail">toggle abstract</button></p>
              <p id="mikhail" class="collapse" style="text-align:left;">
              In this thesis, I focus on methodological concepts of studying infectious disease transmission between agricultural premises. I used different disease systems as exemplars for spatial and network methods to investigate transmission patterns.
              Infectious diseases cause tangible economic threat to the farming industry worldwide by damaging livestock populations, reducing farm productivity and causing trade restriction. This implies the importance of veterinary epidemiological studies in control and eradication of pathogens. Recent increase in availability of data and computational power allowed for more opportunities to study mechanisms of pathogenic transmission. Nowadays, the bottleneck is primarily associated with efficient methods that can analyse vast amounts of high-resolution data.
              Here I address two livestock pathogens that differ in their epidemiology: bacteria Streptococcus agalactiae and foot-and-mouth disease (FMD) virus.
              Streptococcus agalactiae is a contagious pathogen that causes mastitis in cattle, and thus possesses a substantial economic burden to the dairy industry. Known transmission routes between cattle are restricted to those via milking machines, milkers’ hands and fomites during milking process. Additionally, recent studies suggested potential introductions from other host species: primarily, humans. However, strain typing data showed discrepancies in strain compositions of bacteria isolated from humans and bovines. In this thesis, strain-specific features of between-herd transmission of Streptococcus agalactiae within dairy cattle population in Denmark are investigated.
              Foot-and-mouth disease (FMD) is a viral infection that affects cloven-hoofed animals and is of big importance mainly because of the trade restrictions against infected regions and countries. Control programmes against FMD usually include vaccination and culling of animals. However, the debate on the optimal control for FMD is still ongoing. In this thesis, I address questions on identification of the routes of infection and on requirements for movement recording systems to be used for efficient contact tracing during an FMD outbreak.
              This thesis reveals several interesting findings. Firstly, the increased understanding of strain-specific transmission characteristics of Streptococcus agalactiae. One of the observed strains (ST103) showed significant and consistent spatial clustering of its cases among Danish dairy cattle herds in 2009–2011.
              Secondly, the network analysis of cattle movements and affiliations with veterinary practices showed that veterinary practices were exclusively associated with transmission of ST103 of Streptococcus agalactiae. Contrastingly, movement networks appeared to be important for all the three predominant bacterial strains (ST1, ST23 and ST103).
              Fourthly, the new extended approach that allows estimation of the whole transmission tree at once was proposed and tested for the Darlington cluster within the 2001 FMD UK epidemic. Finally, in chapter 6, it was shown that mathematical modelling did not suggest any advantages of ensuring smaller delays in the post-silent control of FMD-like pathogens.
              </p>

              <p><b>Dr. Li (Kevin) Sun</b> Thesis to appear</p>
            </div>
        </div>
      </div>
    </div>
  </div>

  <hr />

  <div class="content-section" id="teaching">
    <div class="container">
      <div class="row">
        <h2>Teaching</h2>
      </div>
      <div class="row">
        <div class="col-md-3">
          <h3>Current modules</h3>
          <p>Machine Learning 4/M - <a href="http://moodle2.gla.ac.uk/course/view.php?id=2847">Moodle</a></p>
          <p>Advanced Programming (IT) - <a href="http://moodle2.gla.ac.uk/course/view.php?id=1928">Moodle</a></p>
          <!-- <p>Systems biology approaches to -omics data analysis - <a href="http://moodle2.gla.ac.uk/course/view.php?id=1928">Moodle</a></p> -->
        </div>
        <div class="col-md-9">
          <h3>Projects</h3>
          <p>I've supervised projects in many areas and I'm happy to discuss opportunities in any area broadly related to my research interests. I'm currently particularly interested in projects in computational metabolomics, ranging from software engineering projects to more open ended research ones. If you would like to see projects from previous years, here are some (email me if you would like one that isn't listed):</p>
          <div>
          <table class="table" id="projecttable">
            <thead>
              <th>Cohort</th><th>Year</th><th>Student</th><th>Title</th>
            </thead>
            <tbody>
              <tr>  
                <td>MSc Bioinformatics</td><td>2012</td><td>Joe Wandy</td><td>...</td>
              </tr>
              <tr>
                <td>MSc</td><td>2015</td><td>Scott Greig</td><td><a href="files/projects/MSc_project_2164953g.pdf" target=new>FrAnK: A Web Application for the Annotation of Mass Spectral Peaks Using Fragmentation Spectra</a></td>
              </tr>
              <tr>
              <td>MSc</td><td>2015</td><td>Mary McDowall</td><td><a href="files/projects/MSc_project_1003254m.pdf" target="new">Structural Comparisons for Small Molecules</a></td>
              </tr>
              <tr>
              <td>MSc</td><td>2015</td><td>James Ferguson</td><td><a href="files/projects/MSc_project_2160915f.pdf" target="new">Investigating the Use of Peak Clustering for the Identification of Metabolites</td>
              </tr>
              <tr>
              <td>MSc</td><td>2015</td><td>Nikolay Zhekov</td><td><a href="files/projects/MSc_project_2163616Z.pdf" target="new">Investigation of different features for predicting ionization efficiency of metabolites in mass spectrometry using Gaussian processes</td>
              </tr>
              <tr>
                <td>MSc</td><td>2015</td><td>Cristina Mihailescu</td><td><a href="files/projects/MSc_project_2168879.pdf" target="new">Implementations of knapsack algorithms for metabolite identification</a></td>
              </tr>
              <tr>
                <td>L4</td><td>2015</td><td>Liam Furphy</td><td><A href="files/projects/L4proj-14-15-FURPHY,Liam.pdf" target="new">Computational Modelling of Red Blood Cells II</A></td>
              </tr>
              <tr>
                <td>L4</td><td>2015</td><td>Daniel Budeanu</td><td><a href="files/projects/Budenau.pdf" target="new">Text Prediction Visualizer for Android</a></td>
              </tr>
              <tr>
                <td>L4</td><td>2015</td><td>Andrew Lau</td><td><a href="files/projects/L4proj-14-15-LAU,andrew.pdf" target="new">A Junior Tennis Diary</a></td>
              </tr>
              <tr> 
                <td>MSci</td><td>2015</td><td>Dimitar Petrov</td><td><a href="files/projects/MSci_project_1005056.pdf" target="new">Improving Touchscreen Typing Using Back-of-Device Grip Interactions</td>
              </tr>
              <tr> 
                <td>MSci</td><td>2015</td><td>Martin Bevc</td><td><a href="files/projects/MSci_project_1006404b.pdf" target="new">Predicting the Outcome of Tennis Matches From Point-by-Point Data</a></td>
              </tr>              
              <tr>
                <td>MSc</td><td>2014</td><td>Josef Idris Khan</td><td><a href="files/projects/MSci_project_8121091k.pdf" target="new">Computational Modelling of Red Blood Cells</a></td>
              </tr>
              <tr>
                <td>MSc</td><td>2014</td><td>Victor Semerdjiev</td><td><a href="files/projects/MSc_project_2106504.pdf" target="new">Machine Learning the Premier League</a></td>
              </tr>
              <tr>
                <td>MSc</td><td>2014</td><td>Jean Georgina Power</td><td><a href="files/projects/MSc_project_2106494P.pdf" target="new">Register Machine Simulator</td>
              </tr>
              <tr>
                <td>L4</td><td>2016</td><td>Adam Kurkiewicz</td><td><a href="files/projects/adam.pdf" target="new">The effect of natural selection on relatedness in randomly mating population</a></td>
              </tr>
              <tr>
                <td>L4</td><td>2016</td><td>Christos Kitsos</td><td><a href="files/projects/Christos.pdf" target="new">Interactive Website For Rabies Researchers</a></td>
              </tr>
              <tr>
                <td>L4</td><td>2016</td><td>Zoe Gerolemou</td><td><a href="files/projects/L4proj-15-16-GEROLEMOU,zoe.pdf" target="new">Automated Genome Informatics Workflow Development</a></td>
              </tr>
              <tr>
                <td>L4</td><td>2016</td><td>Claire Thompson</td><td><a href="files/projects/L4proj-15-16-THOMPSON,claire.pdf" target="new">Porting of a Mass2LDA tool and visualiser into FrAnK (Fragment Annotation Toolkit)</a></td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>

  <hr />

  <div class="content-section" id="fcml">
    <div class="container">
      <div class="row">
        <div class="col-md-2">
          <img vspace=30 width=100% src="images/cover.png">
        </div>
        <div class="col-md-10">
         <h2>A First Course in Machine Learning</h2>
          <h4>Simon Rogers and Mark Girolami, Chapman and Hall / CRC, 2011</h4>
          <p>This introductory level textbook was developed from a Machine Learning course delivered at the University of Glasgow initially by Mark Girolami and then taken over by me. As the course was aimed at final year Computing Science students who may not have taken any courses in statistics or mathematics, the book assumes almost no pre-requisites in mathematics or statistics.</p>
          <p>More details can be found on the book webpage <a href="http://bit.ly/firstcourseml">here</a>.</p>
          <p>It is available direct from the publishers, <a href="http://www.taylorandfrancis.com/books/details/9781439824146/">Chapman and Hall / CRC</a> or from <a href="http://www.amazon.co.uk/Machine-Learning-Chapman-Pattern-Recognition/dp/1439824142/ref=sr_1_1?ie=UTF8&qid=1319973921&sr=8-1">Amazon UK</a> or <a href="http://www.amazon.com/Course-Machine-Learning-Chapman-Hallcrc/dp/1439824142/ref=sr_1_1?ie=UTF8&qid=1319974881&sr=8-1">Amazon US</a>.</p>
        </div>
      </div>
      <div class="row">
        <div class="col-md-10">
          <h2>Second Edition</h2>
          <p>
            The second edition was published in August 2016. This new edition includes three new chapters (>100 pages of new material) that covers some advanced topics: Gaussian Processes, Markov Chain Monte Carlo Sampling, and Dirichlet Processes. When you buy the hard copy you get the e-book version included.
          </p>
        </div>
  
        <div class="col-md-2">
          <img vspcae=30 width=100% src="images/cover2.png">
        </div>
      </div>
    </div>
  </div>

  <hr />

  <div class="content-section" id="direct">
    <div class="container">
      <div class="row">
        <div class="col-md-4">
          <h2>Find me</h2>
          <p>From the foyer of the Sir Alwyn Williams Building (see map on right), either take the stairs to floor 'S' or take the lift to the second floor (use the left hand button; we have two second floors).</p>
          <p>Once on floor 'S', walk along the corridor for about 50m until you come to S123 on the right.</p>
          <p>If you get lost, return to the foyer and ring me using the phone there on extension 1649.</p>
        </div>
        <div class="col-md-8">
          <iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2238.3521537487254!2d-4.293622817578358!3d55.87390578571833!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x488845ce6c50ce15%3A0x9891fb78c4c05809!2sSir+Alwyn+Williams+Bldg%2C+Glasgow%2C+Glasgow+City+G12+8QN!5e0!3m2!1sen!2suk!4v1452610620511" width="100%" height="300" frameborder="0" style="border:0" allowfullscreen></iframe>
        </div>
      </div>
    </div>
  </div>

    <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
    <!-- Include all compiled plugins (below), or include individual files as needed -->
    <script src="js/bootstrap.min.js"></script>


    <script src="js/dataTables/jquery.dataTables.min.js"></script>
    <script src="js/dataTables/dataTables.bootstrap.min.js"></script>

    <script src="js/dataTables/dataTables.buttons.min.js"></script>
    <script src="js/dataTables/buttons.html5.min.js"></script>

    <script type="text/javascript">
      $(document).ready(function() {
        $('#projecttable').DataTable({
        // dom: 'Bfrtip',
        // buttons: [
        //   'csv',
        //   ],
        "order": [[ 1, "desc" ]],
        "iDisplayLength":5,
        "bLengthChange":false,
    });
    });
    </script>

  </body>
</html>